nav2_rrtconnect_planner/include/nav2_rrtconnect_planner/rrt_connect.hpp

```diff
@@ -0,0 +1,130 @@
+/*********************************************************************
+ *
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2008, 2013, Willow Garage, Inc.
+ *  Copyright (c) Shivang Patel
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage, Inc. nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ * Author: Shivang Patel
+ *
+ * Reference tutorial:
+ * https://ros-planning.github.io/navigation2/tutorials/docs/writing_new_nav2planner_plugin.html
+ *********************************************************************/
+#ifndef NAV2_RRTCONNECT_PLANNER_INCLUDE_NAV2_RRTCONNECT_PLANNER_RRT_CONNECT_HPP
+#define NAV2_RRTCONNECT_PLANNER_INCLUDE_NAV2_RRTCONNECT_PLANNER_RRT_CONNECT_HPP
+
+#include <ompl/base/spaces/RealVectorStateSpace.h>
+#include <ompl/base/SpaceInformation.h>
+#include <ompl/base/State.h>
+#include <ompl/geometric/SimpleSetup.h>
+
+#include "rclcpp/rclcpp.hpp"
+#include "geometry_msgs/msg/point.hpp"
+#include "geometry_msgs/msg/pose_stamped.hpp"
+
+#include "nav2_core/global_planner.hpp"
+#include "nav_msgs/msg/path.hpp"
+#include "nav2_util/robot_utils.hpp"
+#include "nav2_util/lifecycle_node.hpp"
+#include "nav2_costmap_2d/costmap_2d_ros.hpp"
+
+
+namespace nav2_rrtconnect_planner
+{
+
+class RRTConnect : public nav2_core::GlobalPlanner
```

Comment:
There is very little that is RRTConnect specific in this class. I would name it OMPLPlanner (or something like that). You can make RRTConnect the default planner, if that seems to work best. You should add a method to set the Planner. Perhaps something like this:
```
template <typename Planner>
void setPlanner()
{
    if (!ss_)
       initializeOMPL();
   ss_->setPlanner(std::make_shared<Planner>(ss_->getSpaceInformation());
}
```

Since you are using cost maps, wouldn't you want to use a planner that is cost-aware? To do so, you first need to create an [ompl::base::OptimizationObjective](http://ompl.kavrakilab.org/classompl_1_1base_1_1OptimizationObjective.html)-derived class that captures the state and/or motion cost of moving over this cost map. Once you have that, try running TRRT or BiTRRT. PRM*/RRT* might also be reasonable choices.

If the path cost is simply distance traveled, then there's no need to set up an OptimizationObjective. This is the default cost that is being optimized when using a cost-aware planner like TRRT, BiTRRT, PRM*, RRT*, ...